lib/SILOptimizer/Differentiation/TangentBuilder.cpp

```diff
@@ -0,0 +1,84 @@
+//===--- TangentBuilder.cpp - Pullback function generation ---*- C++ -*----===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2019 - 2020 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+// This file defines a helper class for emitting tangent code for automatic
+// differentiation.
+//
+//===----------------------------------------------------------------------===//
+
+#include "swift/SILOptimizer/Differentiation/TangentBuilder.h"
+
+namespace swift {
+
+AccessorDecl *TangentBuilder::getAdditiveArithmeticZeroGetter() {
```

Comment:
Ideally this might be cached across instances – seems not a big deal

Reply:
Changed so that `TangentBuilder` holds a reference to `ADContext`.